src/harness/testcases/WINNF_FT_S_WDB_testcase.py

```diff
@@ -0,0 +1,217 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import json
+import logging
+import os
+import sas
+import sas_testcase
+from util import configurable_testcase, writeConfig, loadConfig,\
+    convertRequestToRequestWithCpiSignature, makePalRecordsConsistent
+
+
+class WinnfDatabaseUpdateTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def generate_WDB_1_default_config(self, filename):
+    """Generates the WinnForum configuration for WDB.1"""
+
+    # Load devices info
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+
+    # Load PAL records
+    pal_record_a_0 = json.load(
+      open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+    pal_record_a_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'pal_record_1.json')))
+
+    pal_record_b_0 = json.load(
+      open(os.path.join('testcases', 'testdata', 'pal_record_2.json')))
+    
+    # Set the PAL frequency.
+    pal_low_frequency = 3570000000
+    pal_high_frequency = 3580000000
+
+    # Setting the frequency range and user ID(s) 'U' for PAL records.
+    pal_records_a = makePalRecordsConsistent([pal_record_a_0, pal_record_a_1],
+                                             pal_low_frequency, pal_high_frequency,
+                                             device_a['userId'])
+    pal_records_b = makePalRecordsConsistent([pal_record_b_0],
+                                             pal_low_frequency, pal_high_frequency,
+                                             device_b['userId'])
+
+    # Getting PAL IDs from pal records
+    pal_ids_a = [pal_records_a[0]['palId'], pal_records_a[1]['palId']]
```

Comment:
You need to ensure that these census tracts are adjacent, otherwise the PPA creation request is always expected to fail.

Reply:
updated   the PAL FIPS codes to make records adjucent, referred PCR test case for the   updates